src/services/homework/hooks/submissions.js

```diff
@@ -5,14 +5,50 @@ const hooks = require('feathers-hooks');
 const auth = require('feathers-authentication');
 const errors = require('feathers-errors');
 
+const filterRequestedSubmissions = hook => {
+	// if no db query was given, try to slim down/restrict db request
+	if (Object.keys(hook.params.query).length === 0) {
+		// if user is given
+		//TODO: what if hook.params.account is not set?
+		if (hook.params.account) {
+			let userService = hook.app.service('users');
+			return userService.find({
+				query: {
+					_id: hook.params.account.userId,
+					$populate: ['roles']
+				}
+			}).then(res => {
+				let user = res.data[0];
+				user.roles.map(role => {
```

Comment:
I would rather solve this by permission: VIEW_SUBMISSION_FOR_CLASS and/or VIEW_SUBMISSION_FOR_...

Reply:
I agree with @janrenz